modules/ardrivo/src/Wire.cxx

```diff
@@ -0,0 +1,140 @@
+#include "..\..\..\include\ardpolyfills\Wire.h"
+#include "Wire.h"
+#include "Arduino.h"
+#include "BoardDataDef.hxx"
+
+//Do diffrently??
+bool transmitting = false;
+uint8_t transmissionAdress;
+uint8_t transmitted;
+
+void TwoWire::begin()
+{
+    begun = true;
+}
+
+void TwoWire::begin(uint8_t adress)
+{
+    //Begin to the specific "adress"???
+    begun = true;
+}
+void TwoWire::begin(int adress)
+{
+    //Begin to the specific "adress"???
+
+    begun = true;
+}
+
+size_t TwoWire::requestFrom(uint8_t address, size_t size, bool stop)
+{
+    // ?????
+    // How do you diffrentiate REQUEST_SEND och REQUEST_RECEIVE????
+    /*    
+    REQUEST_SEND = 1,
+    REQUEST_RECEIVE
+    */
+
+    //vad faeeen g�r x??
+    //int x = 0 och x i ic2_buses[x] ????
+
+    //med denna gn�ller den inte
+    ProtocolRequest req = ProtocolRequest(RequestType::REQUEST_SEND, size);
+    transmissionAdress = address;
+    
+    //med denna gn�ller den 
+    //auto req = new ProtocolRequest(RequestType::REQUEST_SEND, size);
+    board_data->i2c_buses[bus_id].requests.at(address).push_back(req);
+
+    // need other return value
+    return 0;
+}
+
+uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+}
+uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size, uint8_t stop)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+}
+uint8_t TwoWire::requestFrom(int address, int size)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+}
+uint8_t TwoWire::requestFrom(int address, int size, int stop)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)stop);
+
+}
+
+void TwoWire::beginTransmission(uint8_t adress)
+{
+    transmitting = true;
+    // set address of targeted slave    
+    transmissionAdress = adress;
```

Comment:
I'd suggest squashing these two variables down to a single `std::optional<Address>`
The optional would be empty when nothing is being transmitted, and would contain an address otherwise